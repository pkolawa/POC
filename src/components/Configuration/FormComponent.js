import React from "react";
import T from 'prop-types';
import { withStyles } from "@material-ui/core/styles";
import FormRow from "./../FormRow";
import { TextField } from "@material-ui/core";

const styles = (theme) => ({

});


/**
 * Allowed field types
 */
export const FORM_FIELD_TYPES = {
	// form fields
    button: 'button',
    number: 'number',
	string: 'string', // text field
	select: 'select', // select
};

class FormComponent extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formFields: []
        }
    }

    static propTypes = {
        pageData: T.array.isRequired,
    };


    renderForm = pageData => {

        return pageData.map(item => {
            const { fields } = item;

            return this._renderFormRow(fields);
        })
    }

    _renderFormRow = rows => {

        return (
            <FormRow
            // header={field.header}
            // {...field.customProps}
            // classes={{}}
            >
                {rows.map(item => {
                    return this.renderSectionItems(item);
                })}
            </FormRow>
        );
    }


    renderSectionItems = (field) => {
        const { type, name, value} = field;
        
        let component;
        switch (type) {
			case FORM_FIELD_TYPES.string:
                component = this.renderTextField(name, value);
                break;
            case FORM_FIELD_TYPES.number:
                component = this.renderNumberField(name, value);
                break;
            default:
                return false
            
        }

        console.log('>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>', component);

        return (
			<React.Fragment key={field.name}>
                {component}
            </React.Fragment>
				
		);
    }

    renderTextField = (name, value) => {
        return <TextField type="text" label={name} value={value} />

    }

    renderNumberField = (name, value) => {
        return <TextField shrink type="number" label={name} value={value} />

    }





    render() {
        const { pageData } = this.props;

        return this.renderForm(pageData);

    }

    componentDidCatch(error, info) {
        console.log('>>>>>>>>>>>>..', error, info);
    }
}

export default withStyles(styles)(FormComponent);
