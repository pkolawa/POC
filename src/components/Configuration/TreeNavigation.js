import React from "react";
import T from 'prop-types';
import { withStyles } from "@material-ui/core/styles";
import TreeView from "@material-ui/lab/TreeView";
import TreeItem from "@material-ui/lab/TreeItem";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import ChevronRightIcon from "@material-ui/icons/ChevronRight";

const styles = (theme) => ({
    
});

class TreeNavigation extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formFields: []
        }
    }

    static propTypes = {
        treeItems: T.array.isRequired,
        nodeClickHandler: T.func.isRequired
	};

    getTreeItemsFromData = treeItems => {
        return treeItems.map(treeItemData => {
            let children = undefined;
            if (treeItemData.children && treeItemData.children.length > 0) {
                children = this.getTreeItemsFromData(treeItemData.children);
            }

            return (
                <TreeItem
                    key={treeItemData.id}
                    nodeId={treeItemData.id}
                    label={this.renderlabel(treeItemData)}
                    children={children}
                />
            );
        });
    };

    renderlabel = item => {
        const { name, id, data } = item;

        return (
            <span onClick={() => this.showDetails(data)} nodeId={id} key={id}>{name}</span>
        )
    }

    showDetails = (data) => {
        const { nodeClickHandler } = this.props;
        nodeClickHandler(data);

    }

    dataTreeView = (treeItems) => {
        return (
            <TreeView
                defaultCollapseIcon={<ExpandMoreIcon />}
                defaultExpandIcon={<ChevronRightIcon />}
            >
                {this.getTreeItemsFromData(treeItems)}
            </TreeView>
        );
    };

    render() {
        const { treeItems } = this.props;
        
        return this.dataTreeView(treeItems);

    }
}

export default withStyles(styles)(TreeNavigation);
