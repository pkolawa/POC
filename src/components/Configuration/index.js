import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Menu from "../Menu";

const styles = (theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    flexDirection: "row",
  },
  container: {
    flexBasis: `calc(100% - 240px)`,
  },
});

class Configuration extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <div className={classes.root}>
        <Menu />
        <Container className={classes.container}>
          <Typography align="center" variant="h2" color="secondary">
            Configuration Page
          </Typography>
        </Container>
      </div>
    );
  }
}

export default withStyles(styles)(Configuration);
