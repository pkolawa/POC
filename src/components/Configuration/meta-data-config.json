[{
  "id": "1",
  "name": "serverpools",
  "children": [
    {
      "id": "2",
      "name": "pools",
      "children": [
        {
          "id": "3",
          "name": "LocalPool",
          "data": [
            {
              "fields": [
                {
                  "name": "id",
                  "type": "string",
                  "value": "local1"
                },
                {
                  "name": "key",
                  "type": "string",
                  "value": "A"
                },
                {
                  "name": "hostname",
                  "type": "string",
                  "value": "loalhost1"
                },
                {
                  "name": "port",
                  "type": "number",
                  "value": "8091"
                }
              ]
            }
          ]
        },
        {
          "id": "4",
          "name": "TestPool"
        }
      ]
    }
  ]
}]