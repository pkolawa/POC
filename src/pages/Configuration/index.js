import React from "react";
import { Helmet } from "react-helmet";
import ConfigurationManagementContainer from "../../containers/ConfigurationManagementContainer";

class Configuration extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <>
        <Helmet>
          <title>Configuration Management</title>
          <meta name="description" content="Configuration Management" />
        </Helmet>
        <ConfigurationManagementContainer />
      </>
    );
  }
}

export default Configuration;
