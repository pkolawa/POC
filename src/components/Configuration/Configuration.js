import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import TreeNavigation from "./TreeNavigation";
import metaData from "./meta-data-config.json";
import FormComponent from "./FormComponent";
import Menu from "../Menu";


const styles = (theme) => ({
    root: {
        display: "flex",
        flexWrap: "wrap",
        flexDirection: "row",
    },
    container: {
        flexBasis: `calc(100% - 240px)`,
    },
});

class Configuration extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formData: [],
        }
    }

    handleNodeClick = (formData) => {
        if(formData && formData.length > 0) {
            this.setState({ formData })
        }
    }

    render() {
        const { classes } = this.props;
        const { formData } = this.state;

        return (
            <div className={classes.root}>
                <Menu />
                <Container className={classes.container}>
                    <Typography align="center" variant="h2" color="secondary">
                        Configuration Page
                    </Typography>
                    <TreeNavigation treeItems={metaData} nodeClickHandler={this.handleNodeClick} />
                    <FormComponent pageData={formData} />
                </Container>
                
            </div>
        );
    }
}

export default withStyles(styles)(Configuration);