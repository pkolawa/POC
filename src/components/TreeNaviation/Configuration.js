import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import TreeNavigation from "./TreeNavigation";
import metaData from "./meta-data-config.json";
import FormComponent from "./FormComponent";
import Menu from "../Menu";

const styles = (theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    flexDirection: "row",
  },
  container: {
    flexBasis: `calc(100% - 240px)`,
  },
  gridContainer: {
      marginTop: theme.spacing(3)
  }
});

class Configuration extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      formData: [],
      // metaData: []
    };
  }

  componentDidMount() {
    //fetch('http://localhost:3001/configuration').then( metaData => this.setState({metaData})).catch(e => console.error(e))
  }

  handleNodeClick = (formData) => {
    if (formData && formData.length > 0) {
      this.setState({ formData });
    }
  };

  render() {
    const { classes } = this.props;
    const { formData } = this.state;

    return (
      <div className={classes.root}>
        <Menu />
        <Container className={classes.container}>
          <Typography align="center" variant="h2" color="secondary">
            Configuration Page
          </Typography>
          <Grid container className={classes.gridContainer} spacing={3}>
            <Grid item xs={3}>
              <TreeNavigation
                treeItems={metaData}
                nodeClickHandler={this.handleNodeClick}
              />
            </Grid>
            <Grid item xs={9}>
              <FormComponent pageData={formData} />
            </Grid>
          </Grid>
        </Container>
      </div>
    );
  }
}

export default withStyles(styles)(Configuration);
