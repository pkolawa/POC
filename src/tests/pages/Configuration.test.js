import React from "react";
import { shallow } from "enzyme";
import Configuration from "../../pages/Configuration";

test("Should Render Configuration Page", () => {
  const wrapper = shallow(<Configuration/>);
  expect(wrapper).toMatchSnapshot();
});