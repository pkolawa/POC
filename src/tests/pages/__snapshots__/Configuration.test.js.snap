// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Configuration Page 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Configuration Management
    </title>
    <meta
      content="Configuration Management"
      name="description"
    />
  </HelmetWrapper>
  <Connect(WithStyles(ConfigurationManagementContainer)) />
</Fragment>
`;
